packages/std/src/ado_contract/rates.rs

```diff
@@ -3,67 +3,75 @@ use crate::amp::Recipient;
 use cw20::Cw20Coin;
 use cw_storage_plus::Map;
 
 use super::ADOContract;
 
-pub fn rates<'a>() -> Map<'a, &'a str, Rate> {
+pub fn rates<'a>() -> Map<'a, &'a str, Vec<Rate>> {
     Map::new("rates")
 }
 
 impl<'a> ADOContract<'a> {
-    /// Sets rates
-    pub fn set_rates(
-        &self,
-        store: &mut dyn Storage,
-        action: impl Into<String>,
-        rate: Rate,
-    ) -> Result<(), ContractError> {
-        let action: String = action.into();
-        self.rates.save(store, &action, &rate)?;
-        Ok(())
-    }
     pub fn execute_rates(
         &self,
         ctx: ExecuteContext,
         rates_message: RatesMessage,
     ) -> Result<Response, ContractError> {
         match rates_message {
-            RatesMessage::SetRate { action, rate } => self.execute_set_rates(ctx, action, rate),
+            RatesMessage::SetRate { action, rates } => self.execute_set_rates(ctx, action, rates),
             RatesMessage::RemoveRate { action } => self.execute_remove_rates(ctx, action),
         }
     }
+    /// Sets rates
+    pub fn set_rates(
+        &self,
+        store: &mut dyn Storage,
+        action: impl Into<String>,
+        rates: Vec<Rate>,
+    ) -> Result<(), ContractError> {
+        let action: String = action.into();
+        self.rates.save(store, &action, &rates)?;
+        Ok(())
+    }
+
     pub fn execute_set_rates(
         &self,
         ctx: ExecuteContext,
         action: impl Into<String>,
-        mut rate: Rate,
+        mut rates: Vec<Rate>,
     ) -> Result<Response, ContractError> {
+        // Ensure the sender is the contract owner
         ensure!(
             Self::is_contract_owner(self, ctx.deps.storage, ctx.info.sender.as_str())?,
             ContractError::Unauthorized {}
         );
-        let action: String = action.into();
-        // Validate rates
-        rate.validate_rate(ctx.deps.as_ref())?;
 
-        let rate = match rate {
-            Rate::Local(ref mut local_rate) => {
+        let action = action.into();
+
+        // Iterate over the rates, validating and updating as needed
+        for rate in &mut rates {
+            // Validate rates
+            rate.validate_rate(ctx.deps.as_ref())?;
+
+            // Update local rates if recipients are empty
+            if let Rate::Local(ref mut local_rate) = rate {
                 if local_rate.recipients.is_empty() {
-                    local_rate.recipients = vec![Recipient::new(ctx.info.sender, None)];
-                    Rate::Local(local_rate.clone())
-                } else {
-                    rate
+                    local_rate
+                        .recipients
+                        .push(Recipient::new(ctx.info.sender.clone(), None));
                 }
             }
-            Rate::Contract(_) => rate,
-        };
-        self.set_rates(ctx.deps.storage, action, rate)?;
+        }
+
+        // Save the updated rates
+        self.set_rates(ctx.deps.storage, action, rates)?;
```

Comment:
This is state breaking, we'd need a migration function for this for any current ADOs